File: archive/ui-affiliate/src/main/frontend/src/app/data/entities/ad.ts
import { Brand } from './brand';

export class Ad {

  public brand: Brand;
  public name: string;
  public guid: string;
  public type: number;
  public entryPoint: string;
  public targetUrl: string;

  public resourceUrlSystem: string;
  public entryPointUrlSystem: string;
  public icon: string;
  public code: string;

  constructor(
    public id: number
    ){}
}

File: service-affiliate/client-service-affiliate/src/main/java/lithium/service/affiliate/client/objects/Ad.java
package lithium.service.affiliate.client.objects;

import java.io.Serializable;
import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Ad  implements Serializable {

	private static final long serialVersionUID = 1L;
	
	public static final int AD_TYPE_LINK = 0;
	public static final int AD_TYPE_IMAGE = 1;
	public static final int AD_TYPE_IFRAME = 2;
	
	private Long id;
	
	int version;

	private String name;

	private String guid;

	private Boolean archived;

	private Boolean deleted;

	private Integer type;

	private Date createdDate;

	private Brand brand;

	private String targetUrl;

	private String entryPoint;
	
}
